### PR TITLE
feat: adds PITR support for volumesnapshot

### LINKED ISSUE #2387: Enable PITR from cold backup snapshots
In 1.20.1 we introduced an imperative way through the `cnpg` plugin to take a cold backup and generate volume snapshot resources. However at the moment, only snapshot recovery is possible (meaning that the database state is brought back up to the exact moment of the backup).

It is not currently possible to attach the recovery to a WAL archive and issue a [Point in Time recovery](https://cloudnative-pg.io/documentation/current/bootstrap/#point-in-time-recovery-pitr).

Add the possibility to execute a PITR from [volume snapshots](https://cloudnative-pg.io/documentation/current/bootstrap/#recovery-from-volumesnapshot-objects).

### PR DESCRIPTION
**Pull request by armru**
This patch introduces PITR support for the volumesnapshots by adding a new instance manager command called 'restoresnapshot'.

Closes #2387

### CODE REVIEW

**Comment by mnencia** [comment]
We need to add support for the pre-restore check using the code in d00f978db3ff911a0da0ed772ed9369ec2310c46

File: tests/e2e/volume_snapshot_test.go [hunk 1]
```diff
@@ -31,76 +32,229 @@ import (
...
+					for _, snapshot := range snapshotList.Items {
+						if strings.Contains(snapshot.Name, snapshotSuffix) {
+							g.Expect(snapshot.Name).To(ContainSubstring(clusterName))
+						}
+					}
+				}).Should(Succeed())
+			})
+		})
+
+		FContext("Can restore from a Volume Snapshot", Ordered, func() {
```
Comment (armru) [inline tests/e2e/volume_snapshot_test.go:113]: this should be removed

Comment (NiccoloFei) [reply tests/e2e/volume_snapshot_test.go]: Done

**Review by leonardoce — APPROVED** [review]

**Review by mnencia — APPROVED** [review]

File: contribute/e2e_testing_environment/README.md [hunk 2]
```diff
@@ -293,8 +293,8 @@ Options supported are:
   is false.
 
 Example:
-1. Trigger an e2e test to run all test cases with `highest` test level, we want to cover most kubernetes and postgres 
-metrics
+1. Trigger an e2e test to run all test cases with `highest` test level, we want
+   to cover most kubernetes and postgres metrics
```
Comment (gbartolini) [inline contribute/e2e_testing_environment/README.md:297]:
```suggestion
   to cover most Kubernetes and Postgres metrics
```

**Review by gbartolini — APPROVED** [review]

**Pull request merged**